dlib/test/global_optimization.cpp

```diff
@@ -232,6 +233,15 @@ namespace
         DLIB_TEST(std::abs(result.x - 1) < 1e-9);
         print_spinner();
 
+        constexpr auto close_enough = -16.0;
+        auto early_result = find_max_global([&early_evals](double x){ early_evals++; return -std::pow(x-2,2.0); }, -10, 1, max_function_calls(10), 0.0, std::vector<function_evaluation>{}, [](double y){ return (y >= close_enough);});
```

Comment:
`close_enough` needs to be captured in the lambda.  I don't know why the tests passed.  But I got an email from appveyor with an error saying visual studio failed to compile because `close_enough` wasn't captured.  But somehow this diff isn't marked as failing any tests.

Reply:
Lambda expressions don't need to explicitly capture constant expressions according to [expr.prim.lambda]/12 because constant expressions are not ODR-used in a lambda according to [basic.def.odr]/3.

Apparently this is an known issue with MSVC fixed in a recent release. https://developercommunity.visualstudio.com/content/problem/367326/problems-with-capturing-constexpr-in-lambda.html

I pushed a fix for this to support older MSVC.

Reply:
Yeah, MSVC has lots of standards compliance problems of this nature :|

Reply:
Huh, I don't see any change from the last time I looked. But appveyor also ran MSVC on this PR again and this time didn't complain, so ¯\_(ツ)_/¯